### DISCUSSION TITLE
[question] Generate Mongoose Schema from Zod Schema

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [colinhacks/zod#2680] [Q&A] [answered]
Hi! Amazing library! I'm promoting it in every opportunity.

Have a question. I'm writing a function that takes a z.object with a zod schema, 
and returns a mongoose.Schema.

I can get the fields with schema.shape, and I can tell if a field is optional using .isOptional().
But how can I get:

1. The default value
2. The runtime type (String, Object, Number, Boolean, Array, etc.)

Any tip will be appreciated. Wishing you health, peace, and success.

**Comment by andreww2012** [comment, marked as answer]
Hello! You can use my library that does exactly what you've described, generates mongoose schemas based on zod schema :)
https://www.npmjs.com/package/mongoose-zod

**Reply by angelhdzmultimedia** [reply]
Will try it tonight definitely. Highly appreciated!!!
